services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,69 @@
 
+import groovy.util.logging.Slf4j
 import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.service.elastic.ElasticQueryBuilder
+import org.openkilda.testing.service.elastic.ElasticService
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def tagName = "kilda-floodlight"
+        def result = elasticService.getLogs(new ElasticQueryBuilder().setTags(tagName).setTimeRange(300).build())
+        assert result?.hits?.total > 0
+
+        then: "There should be discovery messages"
+        result.hits.hits.any {hit -> hit.source.message.toLowerCase().contains("push discovery package via")}
+    }
+
+    def "Check Northbound, Storm and Topology Engine logging"() {
+        when: "A non-existent flow is requested"
+        def flowId = "nonexistentflowid" + System.currentTimeMillis()
+        try {
+            northbound.getFlow(flowId)
+        } catch (HttpClientErrorException e) {
+        }
+        and: "Rules on a switch are validated"
+        def switchId = topologyDefinition.switches[2].dpId
+        northbound.validateSwitchRules(switchId)
+
+        then: "Northbound, Storm and Topology Engine should log these actions within 1 minute"
+        def nbTag = "kilda-northbound"
+        def stormTag = "storm-worker_log"
+        def tpTag = "kilda-tpe"
+
+        boolean nbResult = false
+        boolean stormResult = false
+        boolean tpResult = false
+        int timeout = 60
+
+        Wrappers.wait(timeout, 10) {
+            def nbLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(nbTag).setTimeRange(timeout * 2).
+                    setLevel("ERROR").build())
+            def stormLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(stormTag).
+                    setTimeRange(timeout * 2).setLevel("ERROR").build())
+            def tpLogs = elasticService.getLogs(new ElasticQueryBuilder().setTags(tpTag).setTimeRange(timeout * 2).
+                    setLevel("INFO").build())
+
+            nbResult = nbLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }
+            stormResult = stormLogs?.hits?.hits?.any { hit -> hit.source.message.contains(flowId) }
+            tpResult = tpLogs?.hits?.hits?.any {hit -> hit.source.message.contains(switchId.toString())}
+
+            return nbResult && stormResult && tpResult
+        }
+        assert nbResult: "Northbound should generate an error message about not being able to find a flow"
```

Comment:
This is ok, just FYI you can put those asserts directly into Wrappers.wait and it will report the original failure reason in case of failue
```
Wrappers.wait {
//whatever
assert expression1
assert expression2, "custom message" //will still report this message in case of failure
assert expression3
//note that in this case there is no need to return something. 'wait' will digest a 'null' result as OK
}
```
in this case you can also get rid of lines 46-48

Reply:
Thanks, will use this in next tests